main.go

```diff
@@ -206,50 +187,52 @@ func validateHostPort(hostPort string) error {
 	return nil
 }
 
-func newClient(machines []string, tlsCert, tlsKey, tlsCACert string) (client *etcd.Client) {
-	// set default if not specified in env
-	if len(machines) == 1 && machines[0] == "" {
-		machines[0] = "http://127.0.0.1:4001"
-	}
-	if strings.HasPrefix(machines[0], "https://") {
-		var err error
-		// TODO(miek): machines is local, the rest is global, ugly.
-		if client, err = etcd.NewTLSClient(machines, tlsCert, tlsKey, tlsCACert); err != nil {
-			// TODO(miek): would be nice if this wasn't a fatal error
-			log.Fatalf("skydns: failure to connect: %s", err)
-		}
-		return client
+func newEtcdClient(machines []string, tlsCert, tlsKey, tlsCACert string) (etcd.KeysAPI, error) {
+	etcdCfg := etcd.Config{
+		Endpoints: machines,
+		Transport: newHttpTransport(tlsCert, tlsKey, tlsCACert),
+	}
+	cli, err := etcd.New(etcdCfg)
+	if err != nil {
+		return nil, err
 	}
-	return etcd.NewClient(machines)
+
+	// Periodically sync etcd
+	go cli.AutoSync(ctx, 15*time.Second)
```

Comment:
what does this do? Thought the syncing was needed anymore in new(er) clients. Good be mistaken though.